geoapps/inversion/components/preprocessing.py

```diff
@@ -54,6 +67,9 @@ def window_data(
     :return: Updated data dict.
     :return: Vertices or centroids of the windowed data object.
     """
+    if isinstance(data_object, Grid2D):
```

Comment:
If we want the data_object to be points or subcalsses of points, shouldn't we verify it there?

Reply:
the data_object can be sub-class of BaseObject. We just need to do something special for Grid2D since it doesn't downsample as easily as vertex based objects. The inversion would go through, but were getting a lot of points with Nan at the end.